certbot-apache/certbot_apache/augeasparser.py

```diff
@@ -152,12 +152,21 @@ def find_directives(self, name, exclude=True): # pylint: disable=unused-argument
         return nodes
 
     def find_comments(self, comment, exact=False): # pylint: disable=unused-argument
-        """Recursive search of DirectiveNodes from the sequence of children"""
-        new_metadata = {"augeasparser": self.parser}
-        return [AugeasCommentNode(comment=assertions.PASS,
-                                  ancestor=self,
-                                  filepath=assertions.PASS,
-                                  metadata=new_metadata)]
+        """
+        Recursive search of DirectiveNodes from the sequence of children.
+
+        Note that the argument exact is not implemented here, as it's not
+        currently in use in ApacheConfigurator.
+        """
```

Comment:
also, will we be using exact? when would it not have to match exactly? what was the idea there?

Reply:
`exact = False` allows us to search for partial comments. For example, in ApacheConfigurator `autohsts` enhancement we're adding a generated id for each managed virtualhost, and by doing `exact = False` search allows us to retrieve it. On the contrary, we don't currently have functionality requiring `exact = True`.

Reply:
I added `raise NotImplementedError()` when calling `find_comments()` with `exact = True`

Reply:
if we don't actually need a search for `exact = True`, can we just rip it out?

Reply:
We decided to rip it off from the interface, as there doesn't seem to be any use cases where we would strictly need to do this in the `ParserNode` context, in contrary to letting user to do this in the application.